File: frontend/komsiluk-taxiProject/src/app/core/layout/components/passenger/favorite/services/route.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { RouteCreateDTO, RouteResponseDTO } from '../../../../../../shared/models/route.models';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root',
})
export class RouteService {
  private readonly API = 'http://localhost:8081/api/routes';

  constructor(private http: HttpClient) {}

  findOrCreate(dto: RouteCreateDTO): Observable<RouteResponseDTO> {
    return this.http.post<RouteResponseDTO>(`${this.API}/find-or-create`, dto);
  }
}

File: mobileApplication/app/src/main/java/com/komsiluk/taxi/data/remote/route/RouteService.java
package com.komsiluk.taxi.data.remote.route;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface RouteService {

    @POST("routes/find-or-create")
    Call<RouteResponse> findOrCreate(@Body RouteCreateRequest body);
}
